Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -804,48 +804,210 @@ Another compliant solution is to define the API level to 17 (JELLY_BEAN_MR1) and
 
 
 
-### Testing Object persistance
+### Testing Object Persistence
 
 #### Overview
 
-There are various ways to persistn an object within Android:
+There are various ways to persist an object within Android:
 
 ##### Object Serialization
 
-An object and its data can be represented as a sequence of bytes. In Java, this is possible using object serialization <sup>[1]</sup>. Serialization is not secure by default and is just a binary format or representation that can be used to store data locally as .ser file. It is possible to sign and encrypt serialized data but, if the source code is available, this is always reversible.  
+An object and its data can be represented as a sequence of bytes. In Java, this is possible using object serialization <sup>[1]</sup>. Serialization is not secure by default and is just a binary format or representation that can be used to store data locally as .ser file. It is possible to encrypt and sign/HMAC serialized data as long as the keys are stored safely. Note that, when classes are changed, the `ObjectInputStream` will not be able to create objects from older .ser files.
+The example below shows how to create a `Serializable` class by implementing the `Serializable` interface.
 
-##### Json
+```java
+import java.io.Serializable;
+
+public class Person implements Serializable {
+	private String firstName;
+	private String lastName;
+
+	public Person(String firstName, String lastName) {
+        this.firstName = firstName;
```

Comment:
It seems you mixed tabs and spaces when indenting, which makes the indentation inconsistent.